add project files

nickanely opened the pull request:
**The code automates the process of fetching a daily quote and an image, and then sends this information to a Microsoft Teams channel for team communication or motivation.**

- It uses the requests library to fetch data from specified URLs.
- The PIL library is used to handle and display images.
- The logging library provides detailed information about the process and errors.
- The payload is sent to Teams using a webhook URL, with the message formatted to include both text and image.

nickanely committed:
add project files

SchwarzerWolf commented on __pycache__/content_loader.cpython-311.pyc:
What are those binary files?

SchwarzerWolf commented on content_loader.py:
```diff
@@ -0,0 +1,44 @@
+import requests
+import logging
+import credentials
```
non-pep-compliant import order

SchwarzerWolf commented on content_loader.py:
```diff
@@ -0,0 +1,44 @@
+logger = logging.getLogger(__name__)
+
+
+class ContentLoader:
```
It should be abstract if you meant to create it as an interface

SchwarzerWolf commented on content_loader.py:
```diff
@@ -0,0 +1,44 @@
+        try:
+            response = requests.get(self.quote_url)
+            response.raise_for_status()
+            quote = response.json()['content']
```
Inconsistent quotes style

SchwarzerWolf commented on content_loader.py:
```diff
@@ -0,0 +1,44 @@
+            return quote, author
+        except requests.RequestException as e:
+            logger.error(f"Error loading quote: {e}")
+            return "No quote found."
```
It's better to return None or raise an Exception in this case - it's hard to distinguish actual quote with this kind if text

SchwarzerWolf commented on main.py:
```diff
@@ -0,0 +1,15 @@
+
+    notifier = TeamsNotifier(quote_loader, image_loader)
+
+    notifier.action()
```
bad method name - it should be a verb

SchwarzerWolf commented on main.py:
```diff
@@ -0,0 +1,15 @@
+from content_loader import QuoteLoader, ImageLoader
+from send_to_teams import TeamsNotifier
```
Bad name for a module - if it contains notifiers, call it something like "notifiers"

SchwarzerWolf commented on send_to_teams.py:
```diff
@@ -0,0 +1,92 @@
+from io import BytesIO
+
+from content_loader import ContentLoader
+import credentials
```
non-pep-compliant order import

SchwarzerWolf commented on send_to_teams.py:
```diff
@@ -0,0 +1,92 @@
+            response = requests.get(image_url)
+            response.raise_for_status()
+            image = Image.open(BytesIO(response.content))
+            image.show()
```
Why do you need to show it?

SchwarzerWolf commented on send_to_teams.py:
```diff
@@ -0,0 +1,92 @@
+        self.webhook_url = credentials.WEBHOOK_URL
+
+    def action(self):
+        quote, author = self.quote_loader.load()
```
You might face an unhandled error here in case quote fetching went not well

SchwarzerWolf commented on send_to_teams.py:
```diff
@@ -0,0 +1,92 @@
+class TeamsNotifier:
+    def __init__(self, quote_loader: ContentLoader, image_loader: ContentLoader):
+        self.quote_loader = quote_loader
+        self.image_loader = image_loader
```
Actually, I think that it would be better to have in this class not a logic of fetching your data, but a logic of just sending it. The rest of your work should be done in main script

SchwarzerWolf commented on send_to_teams.py:
```diff
@@ -0,0 +1,92 @@
+        self.image_loader = image_loader
+        self.webhook_url = credentials.WEBHOOK_URL
+
+    def action(self):
```
move your main action to the end if a class

SchwarzerWolf commented on send_to_teams.py:
```diff
@@ -0,0 +1,92 @@
+            ]
+        }
+
+        headers = {
```
It should be a constant, written with a constant notation, and better moved at the  top of your class

SchwarzerWolf commented on send_to_teams.py:
```diff
@@ -0,0 +1,92 @@
+        }
+
+        headers = {
+            'Content-Type': 'application/json'
```
no trailing comma

SchwarzerWolf reviewed (commented):
Good job with class architecture! However, please consider to think how to separate logic of fetching data with logic of sending it to the teams. And please, fix a style)

nickanely committed:
updated code

nickanely commented:
I have updated my code. in latest commit "updated code" there are no binary files.
I have ensured consistent formatting and import ordering.
classes and methods focus on a single responsibility.
Added type hints to methods and functions
Also I have added docstrings so the code is more readable

nbezverkhnii commented on content_loader.py:
```diff
@@ -0,0 +1,85 @@
+        """
+        Abstract method to load content.
+        """
+        raise NotImplementedError("Subclasses must implement this method")
```
There is no need to raise this error manually. If necessary, ABC will do it for you.

nbezverkhnii commented on content_loader.py:
```diff
@@ -0,0 +1,85 @@
+            data = response.json()
+            quote = data.get("content")
+            author = data.get("author")
+            if quote is None or author is None:
```
It is better to add this to the logger as well.

nickanely replied:
fixed issues on another branch, Thanks
